revolv/payments/models.py

```diff
@@ -2,63 +2,253 @@
+        Category of the project begin invested into. We only consider users that
+        have a non-zero pool of investable money.
+        !!! TODO: actually prioritize by Category
+    """
+    amount = models.FloatField()
+    admin = models.ForeignKey(RevolvUserProfile)
+    project = models.ForeignKey("project.Project")
+
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    objects = AdminReinvestmentManager()
+    factories = ImportProxy("revolv.payments.factories", "AdminReinvestmentFactories")
+
+
+class PaymentTypeManager(models.Manager):
+    """
+    Manager for PaymentType.
+    """
+
+    def get_paypal(self, queryset=None):
+        """Return the PaymentTypeManager for paypal payments."""
+        if queryset is None:
+            queryset = super(PaymentTypeManager, self).get_queryset()
+        return queryset.get(name=PaymentType._PAYPAL)
+
+    def get_reinvestment_fragment(self, queryset=None):
+        """Return the PaymentTypeManager for reinvestment_fragment payments."""
         if queryset is None:
-            queryset = super(PaymentInstrumentTypeManager, self).get_queryset()
-        return queryset.get(name=INSTRUMENT_REPAYMENT)
-
-
-class PaymentInstrumentType(models.Model):
-    """
-        Abstraction for a payment instrument. A payment instrument is defined as
-        anything that can be used to create a payment (for example, paypal or any
-        other payment service). Types of payment that the RE-volv app currently
-        supports are as follows:
-
-        1. "paypal": a payment that was made via a credit card processed through
-           the paypal REST sdk. TODO: the distinction between an actual payment
-           through a paypal account and a payment made with credit card but processed
-           by paypal.
-        2. "check": a record or a payment that was made directly to a RE-volv
-           administrator via paper check. These must be entered into the app via
-           the UI in the administrator dashboard page in order for our database to
-           know about them.
-        3. "repayment" (DEPRECATED): this represents a payment that was made from a
-           community organization that RE-volv has already funded and is a return on
-           that solar panel investment. These are available to be reinvested.
-        4. "reinvestment": this is a payment that was allocated to a new project as
-           part the revolving funding model - the money originally came from an organic
-           donation to an already completed project, and then the community repayed
-           revolv the money that is represented in the Payment that has this PaymentInstrumentType.
+            queryset = super(PaymentTypeManager, self).get_queryset()
+        return queryset.get(name=PaymentType._REINVESTMENT)
+
+
+class PaymentType(models.Model):
     """
+    Abstraction for a payment. A payment type is defined as anything that can
```

Comment:
Random typo here (should be abstraction for type of payment, not for payment), I'm pretty sure this is the only issue right now though, so if it is I'll just push the one word fix haha